### PR TITLE
Feature snam arm branch

### PR DESCRIPTION
**Pull request by bartbal**

### CODE REVIEW

File: code/headers/calculate_inverse_kinematics.hpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
+#pragma once
+#include <cmath>
+#include <hwlib.hpp>
+#include <vector3.hpp>
+#define PI 3.14159265
```
Comment (Slimmerik) [inline code/headers/calculate_inverse_kinematics.hpp]: Style guide highlights the use for M_PI

_Math library
Rule: always use the <cmath> library. E.g. use M_PI for pi._

File: code/src/snam_arm.cpp [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    snam_arm_c::snam_arm_c(servo_c servo1, servo_c servo2, servo_c servo3,
+                           servo_c servo4,
+                           calculate_inverse_kinematics_c &calculator)
+        : servo1(servo1),
+          servo2(servo2),
+          servo3(servo3),
+          servo4(servo4),
+          calculator(calculator) {
+        // set arm at start position.
+        servo1.servo_rotate(180, r2d2::robot_arm::angles::theta),
```
Comment (Slimmerik) [inline code/src/snam_arm.cpp]: Magic value "180"

File: code/src/snam_arm.cpp [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        }
+        if (biggest_rotation < gamma_rotation) {
+            biggest_rotation = gamma_rotation;
+        }
+
+        // invert speed.
+        speed = max_speed - speed;
+
+        // get the step size.
+        step = float(degrees_per_speed) / float(speed);
```
Comment (Slimmerik) [inline code/src/snam_arm.cpp]: Use c++style casts

Casting
Rule: always use C++-style casts:

static_cast<long>(x)
dynamic_cast<base_c*>(&derived)

File: code/src/snam_arm.cpp [hunk 4]
```diff
@@ -0,0 +1,215 @@
+#include "snam_arm.hpp"
```
Comment (Slimmerik) [inline code/src/snam_arm.cpp]: #include "snam_arm.hpp" -> #include <snam_arm.hpp>

Voor alle includes wordt <> gebruikt dit staat niet in de style guide maar werd ik persoonlijk tijdens het project op gewezen.

**Review by Slimmerik — CHANGES_REQUESTED** [review]

File: code/headers/calculate_inverse_kinematics.hpp (hunk 1 above)
Comment (bartbal) [reply code/headers/calculate_inverse_kinematics.hpp]: updated this: https://github.com/R2D2-2019/robot_arm/pull/28/commits/27356fd635d511ff4f7e0f665fc40ed23382c826

File: code/src/snam_arm.cpp (hunk 2 above)
Comment (bartbal) [reply code/src/snam_arm.cpp]: I hope it is better now: https://github.com/R2D2-2019/robot_arm/pull/28/commits/43c780193c496d0265c19198e7a5f51c7366e7ac

File: code/src/snam_arm.cpp (hunk 3 above)
Comment (bartbal) [reply code/src/snam_arm.cpp]: changed the casts to static_cast<float>: https://github.com/R2D2-2019/robot_arm/pull/28/commits/d6febe36edf5f4957f12521fe2a0084ec39df187

File: code/src/snam_arm.cpp (hunk 4 above)
Comment (bartbal) [reply code/src/snam_arm.cpp]: I changed all the includes to use <>: https://github.com/R2D2-2019/robot_arm/pull/28/commits/5c7c5881b498883b6b19151a51e7383eedddde70

**Review by Slimmerik — APPROVED** [review]

**Pull request merged**